hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java

```diff
@@ -79,7 +79,7 @@ public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching,
       boolean repeat, boolean updateCacheMetrics) {
     // TODO: is there a hole here, or just awkwardness since in the lruCache getBlock
     // we end up calling l2Cache.getBlock.
-    return lruCache.containsBlock(cacheKey)?
+    return cacheKey.getBlockType().getCategory() == BlockCategory.DATA?
```

Comment:
In CombinedBC, its the NON data block (index/bloom) which goes to onheap LRU cache.  Data blocks go to bucket cache (l2Cache as per variable here)

Reply:
You are right. Have fixed in the last commit.

Reply:
As in branch-1 it is also possible that data blocks are cached in L1, so that it is not a good way to check L1 or L2 according to the block type. I have update the logic in the last commit.